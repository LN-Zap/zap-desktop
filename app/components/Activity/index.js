import Activity from './Activity'

export default Activity
