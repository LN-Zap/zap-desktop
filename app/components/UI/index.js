export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundTertiary from './BackgroundTertiary'
export BackgroundSecondary from './BackgroundSecondary'
export Bar from './Bar'
export Button from './Button'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export DataRow from './DataRow'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export Message from './Message'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export MainContent from './MainContent'
export Menu from './Menu'
export MenuItem from './MenuItem'
export MenuItemGroup from './MenuItemGroup'
export Modal from './Modal'
export Notification from './Notification'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export Page from './Page'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Truncate from './Truncate'
export Value from './Value'
export Wizard from './Wizard'
