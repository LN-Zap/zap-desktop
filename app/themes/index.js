import dark from './dark'
import light from './light'

export { dark, light }
