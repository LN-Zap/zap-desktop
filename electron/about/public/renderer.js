'use strict'
Object.defineProperty(exports, '__esModule', { value: true })

window.ipcRenderer.on('about-window:info', (_, info) => {
  const app_name = info.product_name
  const content = info.use_inner_html ? 'innerHTML' : 'innerText'
  document.title = `About ${app_name}`
  const title_elem = document.querySelector('.title')
  title_elem.innerText = `${app_name}`

  const copyright_elem = document.querySelector('.copyright')
  if (info.copyright) {
    copyright_elem[content] = info.copyright
  } else if (info.license) {
    copyright_elem[content] = `Distributed under ${info.license} license.`
  }
  const icon_elem = document.getElementById('app-icon')
  icon_elem.src = info.icon_path
  if (info.description) {
    const desc_elem = document.querySelector('.description')
    desc_elem[content] = info.description
  }

  if (info.css_path) {
    const css_paths = Array.isArray(info.css_path) ? info.css_path : [info.css_path]
    for (const css_path of css_paths) {
      const link = document.createElement('link')
      link.rel = 'stylesheet'
      link.href = css_path
      document.head.appendChild(link)
    }
  }

  if (info.use_version_info) {
    const versions = document.querySelector('.versions')
    const vs = process.versions
    for (const name of ['electron', 'chrome', 'node', 'v8']) {
      const tr = document.createElement('tr')
      const name_td = document.createElement('td')
      name_td.innerText = name
      tr.appendChild(name_td)
      const version_td = document.createElement('td')
      version_td.innerText = ' : ' + vs[name]
      tr.appendChild(version_td)
      versions.appendChild(tr)
    }
  }
})
//# sourceMappingURL=renderer.js.map
