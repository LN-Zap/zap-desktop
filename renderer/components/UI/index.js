export ActionBar from './ActionBar'
export ActionButton from './ActionButton'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export { BasicInput } from './Input'
export { BasicRange } from './Range'
export { BasicTextArea } from './TextArea'
export { BasicCheckbox } from './Checkbox'
export { BasicSelect } from './Select'
export Button from './Button'
export Card from './Card'
export Checkbox from './Checkbox'
export CloseButton from './CloseButton'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export DialogOverlay from './DialogOverlay'
export Donut from './Donut'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export LndConnectionStringInput from './LndConnectionStringInput'
export LndConnectionStringEditor from './LndConnectionStringEditor'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export { ModalOverlayStyles } from './Modal'
export NodePubkeyInput from './NodePubkeyInput'
export Notification from './Notification'
export OpenDialogButton from './OpenDialogButton'
export OpenDialogInput from './OpenDialogInput'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export ProgressBar from './ProgressBar'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioCard from './RadioCard'
export RadioGroup from './RadioGroup'
export Range from './Range'
export RowLabel from './RowLabel'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Tab from './Tab'
export Tabs from './Tabs'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Tooltip from './Tooltip'
export TransactionFeeInput from './TransactionFeeInput'
export Value from './Value'
export Wizard from './Wizard'
