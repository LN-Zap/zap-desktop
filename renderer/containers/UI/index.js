export CurrencyFieldGroup from './CurrencyFieldGroup'
export CryptoSelector from './CryptoSelector'
export CryptoValue from './CryptoValue'
export FiatSelector from './FiatSelector'
export FiatValue from './FiatValue'
