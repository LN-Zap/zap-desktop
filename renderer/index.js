import React from 'react'

import jstz from 'jstimezonedetect'
import ReactDOM from 'react-dom'
import { Provider } from 'react-intl-redux'

import { getDefaultLocale } from '@zap/i18n'
import translations from '@zap/i18n/translation'

import Root from './containers/Root'
import { configureStore, history } from './store/configureStore'

// Default the locale to English.
const defaultLocale = getDefaultLocale()

// Initialise the intl store with data from the users current locale.
const initialState = {
  intl: {
    locale: defaultLocale,
    messages: translations[defaultLocale],
    timeZone: jstz.determine().name(),
  },
}

// Set up the redux store.
const store = configureStore(initialState)
const MOUNT_NODE = document.getElementById('root')

const render = Component => {
  ReactDOM.render(
    <Provider store={store}>
      <Component history={history} />
    </Provider>,
    MOUNT_NODE
  )
}

render(Root)

if (module.hot) {
  module.hot.accept('./containers/Root', () => {
    render(Root)
  })
}
