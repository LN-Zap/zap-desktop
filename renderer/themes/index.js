import dark from './dark'
import light from './light'

export { dark }
export { light }

export default { dark, light }
