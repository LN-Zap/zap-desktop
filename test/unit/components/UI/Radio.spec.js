import React from 'react'
import renderer from 'react-test-renderer'
import { dark } from 'themes'
import { ThemeProvider } from 'styled-components'
import { Form, Radio, RadioGroup } from 'components/UI'

describe('component.UI.Radio', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={dark}>
          <Form>
            <RadioGroup field="radio">
              <Radio value="item1" label="Item 1" description="Radio buttons" />
              <Radio value="item2" label="Item 2" description="can have an optional title" />
              <Radio value="item3" label="Item 3" description="and description" />
            </RadioGroup>
          </Form>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
