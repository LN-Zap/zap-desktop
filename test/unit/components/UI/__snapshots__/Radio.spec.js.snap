// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Radio should render correctly 1`] = `
.c2 {
  font-size: 13px;
  color: #ffffff;
  line-height: 1.4;
}

.c3 {
  margin-top: 8px;
  font-size: 13px;
  color: #959595;
  line-height: 1.4;
}

.c1 {
  margin-bottom: 16px;
  width: 100%;
  font-size: 13px;
  color: #ffffff;
  color: #ffffff;
  font-weight: 300;
  display: block;
}

.c0 .container {
  display: block;
  position: relative;
  padding-left: 30px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.c0 .selection {
  position: absolute;
  top: 1px;
  left: 0;
  height: 16px;
  width: 16px;
  border: 1px solid #959595;
  border-radius: 50%;
}

.c0 .container:hover input ~ .selection {
  border: 1px solid #fd9800;
}

.c0 .container input:checked ~ .selection {
  border: 1px solid #fd9800;
}

.c0 .selection:after {
  content: '';
  position: absolute;
  display: none;
}

.c0 .container input:checked ~ .selection:after {
  display: block;
}

.c0 .container .selection:after {
  top: 3px;
  left: 3px;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  background: #fd9800;
}

<form
  className=""
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="c0"
  >
    <label
      className="container c1"
      color="primaryText"
      display="block"
      fontSize="m"
      fontWeight="light"
      htmlFor="item1"
      opacity={null}
      width={1}
    >
      <div
        className="c2"
        color="primaryText"
        fontSize="m"
      >
        Item 1
      </div>
      <input
        checked={false}
        id="item1"
        onBlur={[Function]}
        onChange={[Function]}
        type="radio"
        value="item1"
      />
      <span
        className="selection"
      />
      <div
        className="c3"
        color="gray"
        fontSize="m"
      >
        Radio buttons
      </div>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="container c1"
      color="primaryText"
      display="block"
      fontSize="m"
      fontWeight="light"
      htmlFor="item2"
      opacity={null}
      width={1}
    >
      <div
        className="c2"
        color="primaryText"
        fontSize="m"
      >
        Item 2
      </div>
      <input
        checked={false}
        id="item2"
        onBlur={[Function]}
        onChange={[Function]}
        type="radio"
        value="item2"
      />
      <span
        className="selection"
      />
      <div
        className="c3"
        color="gray"
        fontSize="m"
      >
        can have an optional title
      </div>
    </label>
  </div>
  <div
    className="c0"
  >
    <label
      className="container c1"
      color="primaryText"
      display="block"
      fontSize="m"
      fontWeight="light"
      htmlFor="item3"
      opacity={null}
      width={1}
    >
      <div
        className="c2"
        color="primaryText"
        fontSize="m"
      >
        Item 3
      </div>
      <input
        checked={false}
        id="item3"
        onBlur={[Function]}
        onChange={[Function]}
        type="radio"
        value="item3"
      />
      <span
        className="selection"
      />
      <div
        className="c3"
        color="gray"
        fontSize="m"
      >
        and description
      </div>
    </label>
  </div>
</form>
`;
